Add 42NAB (#397)
neo4j/docs-status-codes: modules/ROOT/pages/errors/gql-errors/42NAB.adoc

= 42NAB

== Status description
error: syntax error or access rule violation - not supported standalone call. WHERE is not supported in standalone call. Use `CALL ... YIELD ... WHERE ... RETURN ...` instead.

== Explanation
A standalone call of a procedure does not support a WHERE clause, i.e. it does not support the filtering of the procedure result.

> Either remove the WHERE clause or add a RETURN clause after the call to it into an in-query call.

Author: There was a verb missing on my part. What I meant to have it this:

```suggestion
Either remove the `WHERE` clause or add a `RETURN` clause after the call to turn it into an in-query call.
```

== Example scenario
For example, trying to filter the result of the procedure `db.labels()` in a standalone call:

[source,cypher]
----
CALL db.labels() YIELD label WHERE label <> 'A'
----

An error will be thrown with GQLSTATUS 42NAB and the following status description:

[source]
----
error: syntax error or access rule violation - not supported standalone call. WHERE is not supported in standalone call. Use `CALL ... YIELD ... WHERE ... RETURN ...` instead.
----

== Possible solutions

You can abstain from filtering to get a valid query:

[source,cypher]
----
CALL db.labels() YIELD label
----

> However, to filter the result of the procedure `db.labels()` use an in-query call by adding a RETURN clause:

Reviewer: ```suggestion
However, if you still want to filter the result of the procedure `db.labels()`, you can use an in-query call to add a `RETURN` clause:
```

Author: The addition of a RETURN clause is what makes it an in-query call. But we side step that and simply say:

```suggestion
However, if you still want to filter the result of the procedure `db.labels()`, you add a `RETURN` clause:
```

That okay?
